src/TorchSharp/Tensor/Storage.cs

```diff
@@ -0,0 +1,522 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Runtime.InteropServices;
+using System.Diagnostics;
+using System.Linq;
+using System.Runtime.CompilerServices;
+
+namespace TorchSharp
+{
+    public static partial class torch
+    {
+        /// <summary>
+        /// A torch.Storage is a contiguous, one-dimensional array of a single data type.
+        /// Every tensor has a corresponding storage of the same data type.
+        /// </summary>
+        public abstract class Storage
+        {
+            internal Storage()
+            {
+            }
+
+            internal Storage(Tensor tensor)
+            {
+                _tensor = tensor;
+                data_ptr();
+            }
+
+            internal Storage(Tensor tensor, IntPtr data_ptr)
+            {
+                _tensor = tensor;
+                _tensor_data_ptr = (data_ptr == IntPtr.Zero) ? this.data_ptr() : data_ptr;
+            }
+
+            internal static Storage<T> CreateTypedStorageInstance<T>(Tensor tensor) where T : unmanaged
```

Comment:
>CreateTypedStorageInstance [](http://example.com/codeflow?start=39&length=26)

nit: I wouldn't include the term `Typed` in the method name as the method is already a generic method which indicate that.